network/statsd-socket.test.js

```diff
@@ -0,0 +1,346 @@
+jest.mock('./socket', () => ({ Socket: jest.fn() }));
+const { Socket } = require('./socket');
+const { StatsdSocket } = require('./statsd-socket');
+
+describe('StatsdSocket', () => {
+  describe('constructor', () => {
+    it('should create socket with given host and default values', () => {
+      // eslint-disable-next-line no-new
+      new StatsdSocket({
```

Comment:
so you still want to use constructor for this function ?

Reply:
In the meantime yes - I don't want to confuse different standards in the repo. We will change it as a single PR all across the repository (its backward compatible so its not a must for version 1)